CRM/Contact/BAO/Group.php

```diff
@@ -94,6 +96,18 @@ public static function discard($id) {
     $query = "DELETE FROM civicrm_acl_entity_role where entity_table = 'civicrm_group' AND entity_id = %1";
     CRM_Core_DAO::executeQuery($query, $params);
 
+    //check whether this group contains  any saved searches and check if that saved search is appropriate to delete.
+    $groupDetails = Group::get(FALSE)->addWhere('id', '=', $id)->execute();
```

Comment:
@seamuslee001 no need to change this but I generally add ->first() at the end & then don't need to refer with [0] in the next rows